client/state/billing-transactions/transaction-filters/test/reducer.js

```diff
@@ -0,0 +1,153 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import {
+	BILLING_TRANSACTIONS_FILTER_SET_APP,
+	BILLING_TRANSACTIONS_FILTER_SET_DATE,
+	BILLING_TRANSACTIONS_FILTER_SET_PAGE,
+	BILLING_TRANSACTIONS_FILTER_SET_QUERY,
+} from 'state/action-types';
+import { app, date, page, query } from 'state/billing-transactions/transaction-filters/reducer';
+
+describe( 'transaction filter reducer', () => {
+	describe( '#app', () => {
+		test( 'should update from empty filter', () => {
+			const state = app( '', {
+				type: BILLING_TRANSACTIONS_FILTER_SET_APP,
+				transactionType: 'past',
+				app: 'Test app',
+			} );
+			expect( state ).toEqual( 'Test app' );
+		} );
+
+		test( 'should update existing filter', () => {
+			const state = app( 'Test app', {
+				type: BILLING_TRANSACTIONS_FILTER_SET_APP,
+				transactionType: 'past',
+				app: 'Another test app',
+			} );
+			expect( state ).toEqual( 'Another test app' );
+		} );
+
+		test( 'should clear the filter', () => {
+			const state = app( 'Test app', {
+				type: BILLING_TRANSACTIONS_FILTER_SET_APP,
+				transactionType: 'past',
+				app: '',
+			} );
+			expect( state ).toEqual( '' );
+		} );
+	} );
+
+	describe( '#date', () => {
+		test( 'should set the date filter to month', () => {
+			const state = date( Object.freeze( { newest: true } ), {
```

Comment:
We usually use `deepFreeze` for freezing, would you like to apply it here for consistency? Totally up to you, as this will also work for this flat object.